src/main/java/org/wildfly/security/http/HttpAuthenticator.java

```diff
@@ -152,7 +150,7 @@ private SecurityIdentity login(String username, String password, String mechanis
         } catch (IllegalArgumentException | RealmUnavailableException | IllegalStateException e) {
             httpExchangeSpi.authenticationFailed(e.getMessage(), mechanismName);
         } finally {
-            evidence.destroy();
+            if (evidence != null) evidence.destroy();
```

Comment:
Again I consider this to be an anti-pattern for things that need cleanup.

Reply:
The problem is to avoid this we need a double try / finally which would re-introduce the whitespace changes commented on in the first PR - although as this is a SPI method maybe instead of receiving a password it should receive an Evidence instance.  That may actually make more sense.

Reply:
Whitespace-only changes are fine IMO, as you can use the ?w=1 trick in github or ``git diff -w ...`` on the command line.